### ISSUE TITLE
Fix lifetime parameters in icu_uniset::props APIs

### ISSUE THREAD

**Issue by sffc** [unicode-org/icu4x#1095]
Some of the icu_uniset::props APIs take a `'data` data provider, like:

```
(provider: &'data D)
```

But `'data` should not be on the reference to `D`.

**Comment by Manishearth** [comment]
I think this was also fixed by #1297

**Comment by sffc** [comment]
Verified that #1297 has the following change:

```
- pub fn get_for_script<'data, D>(provider: &'data D, enum_val: Script) -> UnisetResult
+ pub fn get_for_script<D>(provider: &D, enum_val: Script) -> UnisetResult
```
